upload/src/addons/SV/CustomFieldPerms/Setup.php

```diff
@@ -6,10 +6,47 @@
 use XF\AddOn\StepRunnerInstallTrait;
 use XF\AddOn\StepRunnerUninstallTrait;
 use XF\AddOn\StepRunnerUpgradeTrait;
+use XF\Db\Schema\Alter;
+use XF\Db\Schema\Create;
 
 class Setup extends AbstractSetup
 {
 	use StepRunnerInstallTrait;
 	use StepRunnerUpgradeTrait;
 	use StepRunnerUninstallTrait;
+
+	public static $tables1 = array(
+		'xf_user_field' => array(
+			'sedo_perms_input_enable'		=> array('type'=>'tinyint unsigned', 'default'=>0, 'entity_type'=>\XF\Mvc\Entity\Entity::UINT, 'entity_default'=>0, 'field_type'=>'uint'),
+			'sedo_perms_output_pp_enable'	=> array('type'=>'tinyint unsigned', 'default'=>0, 'entity_type'=>\XF\Mvc\Entity\Entity::UINT, 'entity_default'=>0, 'field_type'=>'uint'),
+			'sedo_perms_output_ui_enable'	=> array('type'=>'tinyint unsigned', 'default'=>0, 'entity_type'=>\XF\Mvc\Entity\Entity::UINT, 'entity_default'=>0, 'field_type'=>'uint'),
+			'sedo_perms_input_val'			=> array('type'=>'blob', 'entity_type'=>\XF\Mvc\Entity\Entity::SERIALIZED, 'entity_default'=>'', 'field_type'=>'array'),
+			'sedo_perms_output_pp_val'		=> array('type'=>'blob', 'entity_type'=>\XF\Mvc\Entity\Entity::SERIALIZED, 'entity_default'=>'', 'field_type'=>'array'),
+			'sedo_perms_output_ui_val'		=> array('type'=>'blob', 'entity_type'=>\XF\Mvc\Entity\Entity::SERIALIZED, 'entity_default'=>'', 'field_type'=>'array'),
```

Comment:
The blob types need to be default null, or it will break creating users when the add-on is disabled or upgrading.

Note; isset(null) returns false. array_key_exists is what you need to use.

Reply:
MySQL says a blob can't have a default: https://dev.mysql.com/doc/refman/5.7/en/blob.html

Reply:
You need to allow it to be nullable (XF doesn't do nullable by default), at which point inserts into a row WITHOUT a value for that column will be set to null.

Additionally; Most variants of MySQL also allow setting "null" as a default value for text/blob since null is "not a value", so by the letter of the documentation it is correct.

 It is kinda dumb.